tool2schema/parameter_schema.py

```diff
@@ -100,184 +89,20 @@ def to_json(self) -> dict:
-        """
-        Convert the given value to its JSON representation. Overriding methods should
-        also override `decode_value` to convert the value back to its original type.
-
-        :param value: The value to be converted
-        :return: The JSON representation of the value
-        """
-        return value
-
-    def decode_value(self, value):
-        """
-        Convert the given value from the JSON representation to an instance
-        that can be passed to the original method as a parameter. Overriding
-        methods should check whether the value needs to be converted, and return
-        it as is if no conversion is necessary.
-
-        :param value: The value to be converted
-        :return: An instance of the type required by the original method
-        """
-        return value
-
-
-class ValueTypeSchema(ParameterSchema):
-    """
-    Parameter schema for value types.
-    """
-
-    @staticmethod
-    def matches(parameter: Parameter) -> bool:
-        return parameter.annotation != Parameter.empty and parameter.annotation.__name__ in TYPE_MAP
-
-    def _get_type(self) -> Union[str, Parameter.empty]:
-        return TYPE_MAP[self.parameter.annotation.__name__]
-
-
-class GenericParameterSchema(ParameterSchema):
-    """
-    Base class for generic parameter types supporting subscription.
-    """
-
-    def _get_sub_type(self) -> Union[str, Parameter.empty]:
-        if args := typing.get_args(self.parameter.annotation):
-            return TYPE_MAP.get(args[0].__name__, "object")
-
-        return Parameter.empty
-
-
-class ListTypeParameterSchema(GenericParameterSchema):
-    """
-    Parameter schema for list (array) types.
-    """
-
-    @staticmethod
-    def matches(parameter: Parameter) -> bool:
-        return parameter.annotation != Parameter.empty and (
-            parameter.annotation is list or typing.get_origin(parameter.annotation) is list
-        )
-
-    def _get_type(self) -> Union[str, Parameter.empty]:
-        return TYPE_MAP["list"]
-
-    def _get_items(self) -> Union[str, Parameter.empty]:
-        if (sub_type := super()._get_sub_type()) != Parameter.empty:
-            return {"type": sub_type}
-
-        return Parameter.empty
-
-
-class OptionalTypeParameterSchema(GenericParameterSchema):
-    """
-    Parameter schema for typing.Optional types.
-    """
-
-    @staticmethod
-    def matches(parameter: Parameter) -> bool:
-        args = typing.get_args(parameter.annotation)
-        return (
-            parameter.annotation != parameter.empty
-            and typing.get_origin(parameter.annotation) is Union
-            and len(args) == 2
-            and type(None) in args
-        )
-
-    def _get_type(self) -> Union[str, Parameter.empty]:
-        return super()._get_sub_type()
-
 
 class EnumParameterSchema(ParameterSchema):
     """
-    Parameter schema for enumeration types.
+    Parameter schema for enumeration types manually added via add_enum.
     """
```

Comment:
Why not have this in the `ParameterSchema` class as a `create_enum` method?

Reply:
Sure, I added a `add_enum` method in ParameterSchema which changes the type:
``` python
def add_enum(self, values: list) -> None:
        self.type_schema = EnumTypeSchema(values)
```
3150df2